src/prefix_tf_republisher_node.cpp

```diff
@@ -0,0 +1,124 @@
+#include <hector_multi_robot_tools/prefix_tf_republisher_node.hpp>
+
+namespace hector_multi_robot_tools {
+PrefixTfRepublisherNode::PrefixTfRepublisherNode() : PrefixTfRepublisherNode(rclcpp::NodeOptions()) {
+}
+
+PrefixTfRepublisherNode::PrefixTfRepublisherNode(const rclcpp::NodeOptions &options)
+    : rclcpp::Node("prefix_tf_republisher_node", options)
+{
+    if (get_effective_namespace() == "/") {
+        RCLCPP_FATAL_STREAM(get_logger(), "Node is started in global namespace. This will lead to a topic loop.");
+        rclcpp::shutdown();
+    }
+
+    rcl_interfaces::msg::ParameterDescriptor frame_prefix_descriptor;
+    frame_prefix_descriptor.description = "Frame prefix to prepend on frame_id (default: node namespace)";
+    frame_prefix_descriptor.read_only = true;
+    std::string default_frame_prefix = stripLeadingSlash(get_effective_namespace());
+    frame_prefix_ = declare_parameter("frame_prefix", default_frame_prefix);
+    RCLCPP_INFO_STREAM(get_logger(), "Republishing tf frame ids with prefix '" << frame_prefix_ << "'");
+
+    const std::string input_topic = "tf";
+    std::optional<rclcpp::QoS> qos;
+    rclcpp::Rate rate(10);
+    while (!qos) {
+        rate.sleep();
+        RCLCPP_INFO_STREAM_THROTTLE(get_logger(), *get_clock(), 1, "Waiting for input topic QoS profile.");
+        qos = tryDiscoverQoSProfile(input_topic);
+    }
+
+    pub_ = create_publisher<tf2_msgs::msg::TFMessage>("/tf", qos.value());
```

Comment:
Why do you use the worst QoS from the tf publishers to republish instead of using a default?
It makes sense that you have to be on the subscriber side, but the publisher is independent of that, isn't it?

Reply:
It tries to replicate the QoS profile of publishers. If they do not agree, the default `best_effort()` and `durability_volatile()` is used. Or do you mean the duration-based policies?

Reply:
To ensure that a connection can be made, the "worst" available option must be used, see [QoS compatibilities](https://docs.ros.org/en/jazzy/Concepts/Intermediate/About-Quality-of-Service-Settings.html#qos-compatibilities).

Reply:
That's only true on the subscriber side not for the publisher.

Reply:
Yes, but why would you want a different setting for the publisher if this is pretty much a relay node?

Reply:
Currently, if one publisher on tf_static doesn't use transient local and reliable, your republished would not use it at all.
That is a bit construed, but it shows that this is not the most general or robust solution.
I think the most general one would be to let the user specify the QoS.